crates/sl-mpc-mate/src/math.rs

```diff
@@ -75,51 +90,68 @@ where
     ///
     /// `x`: point at which to compute the derivative.
     /// Arithmetic is done modulo the curve order
-    pub fn derivative_at(&self, n: usize, x: &C::Scalar) -> C::Scalar {
+    pub fn derivative_at(&self, n: usize, x: &G::Scalar) -> G::Scalar
+    where
+        G::Scalar: Reduce<U256>,
+    {
         self.coeffs
             .iter()
             .enumerate()
             .skip(n)
             .map(|(i, coeff)| {
                 // TODO build static table of factorials ??
                 //      U256::wrapping_mul if const fn
-                let num: C::Uint = factorial_range(i - n, i); //
-                let scalar_num = C::Scalar::reduce(num);
+                let num: U256 = factorial_range(i - n, i); //
```

Comment:
Could we implement `factorial_range` in terms of `G::Scalar` instead of `U256` ?